### PR TITLE
Add Product Scenario TestCases for Manage Roles with SCIM 2.0

### PR DESCRIPTION
**Pull request by sarubi**

### CODE REVIEW

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/pom.xml [hunk 1]
```diff
@@ -30,4 +30,137 @@
     <name>Manage roles with SCIM 2.0</name>
     <packaging>jar</packaging>
 
+
```
Comment (thanujalk) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/pom.xml]: remove the additional line

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#
+# Copyright 2018 WSO2, Inc. (http://wso2.com)
```
Comment (thanujalk) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/log4j.properties]: 2019

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/testng.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
```
Comment (thanujalk) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/testng.xml]: add wso2 license header.

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/pom.xml (hunk 1 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/pom.xml]: removed

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/testng.xml (hunk 3 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/testng.xml]: Fixed

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/log4j.properties (hunk 2 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/resources/log4j.properties]: fixed

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesSCIM2TestCase.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    }
+
+    @DataProvider(name = "manageRolesConfigProvider")
+    private static Object[][] manageRolesConfigProvider() throws Exception {
+
+        return new Object[][]{
+                {
+                        ADMIN_USERNAME, ADMIN_PASSWORD, SUPER_TENANT_DOMAIN, "scim2Group1.json"
+                },
```
[the remark this answers is not in the document]
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesSCIM2TestCase.java]: formatted

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RoleNamingViolationSCIM2TestCase.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.identity.scenarios.commons.SCIM2CommonClient;
+import org.wso2.identity.scenarios.commons.ScenarioTestBase;
+
+import static org.testng.Assert.assertEquals;
+import static org.wso2.identity.scenarios.commons.util.Constants.IS_HTTPS_URL;
+
+public class RoleNamingViolationSCIM2TestCase extends ScenarioTestBase {
```
Comment (pamodaaw) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RoleNamingViolationSCIM2TestCase.java]: Shall we add a comment for the class and all the other test classes in this PR

Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RoleNamingViolationSCIM2TestCase.java]: addressed with https://github.com/wso2/product-is/pull/6381

**Pull request merged**